### PR TITLE
sql: async writes to system eventlog

### PR DESCRIPTION
**Pull request by knz**
Informs #73308.
Epic: CRDB-18596

This PR aims to remove the synchronous write to system.eventlog for structured events, which is undesirable:

-   it adds a 5-way write to many small "maintainance" transactions.
-   it forces a global roundtrip in multi-region schema changes.

Instead it makes the writes asynchronous.

This PR subsumes #84983.

See individual commits for details.

Release justification: bug fixes

### CODE REVIEW

**Comment by knz** [comment]
@andreimatei could you review this in replacement to #84983?

It's simpler. If folk like it, we can stop the deprecation process for system.eventlog.

**Comment by knz** [comment]
FWIW, with the async write enabled, `cockroach demo` is visibly faster to start already.

**Review by andreimatei — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/86174)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @knz)

___
*[`-- commits` line 4 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YZt-d9KBpNhP7-P8T):*
is it really?
___
*[`-- commits` line 19 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YZun98oQDNL6k6hv_):*
is it really?
___
*[`-- commits` line 42 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YcOwt0PCzvXyXqwv5):*
"cluster setting"
___
*[`-- commits` line 49 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ycdim1Sswr0g7deoK):*
do we really need this cluster setting? I'd vote against it.
___
*[`pkg/sql/event_log.go` line 128 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Y_5NjBKmfvPstiE4v:-N9Y_5NjBKmfvPstiE4w:b-k6h45m) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b7f0058325de8b8b321a08137d462c418c3047f/pkg/sql/event_log.go#L128)):*
> ```Go
> 	// For DDL statements, this is typically the ID of
> 	// the affected descriptor.
> 	targetID int32
> ```

please put a comment somewhere around the schema definition for `system.eventlog` saying that the respective column is not populated any more (or is populated with 0)
___
*[`pkg/sql/event_log.go` line 286 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YalJi08MwqjCu2QHW:-N9YalJi08MwqjCu2QHX:b-fg0sv5) ([raw file](https://github.com/cockroachdb/cockroach/blob/f8a8de03713e02a3a0b7039908209054316ffc72/pkg/sql/event_log.go#L286)):*
> ```Go
> 		prevDescID := m.DescriptorID
> 		*m = commonSQLEventDetails
> 		if m.DescriptorID == 0 {
> ```

I don't really understand what's going on here. Perhaps put a method comment explaining what should and should not be populated in `commonSQLEventDetails`.
___
*[`pkg/sql/event_log.go` line 534 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Y__bDDNU8PYv-HFh5:-N9Y__bDDNU8PYv-HFh6:bb7d3ls) ([raw file](https://github.com/cockroachdb/cockroach/blob/f8a8de03713e02a3a0b7039908209054316ffc72/pkg/sql/event_log.go#L534)):*
> ```Go
> 	reportingID := ex.s.cfg.NodeInfo.NodeID.SQLInstanceID()
> 	const colsPerEvent = 5
> 	// Note: we insert the alue zero as targetID because sadly this
> ```

alue
___
*[`pkg/sql/event_log.go` line 534 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Y_ne_2748fjgM7Van:-N9Y_ne_2748fjgM7Vao:bh997sq) ([raw file](https://github.com/cockroachdb/cockroach/blob/f8a8de03713e02a3a0b7039908209054316ffc72/pkg/sql/event_log.go#L534)):*
> ```Go
> 	reportingID := ex.s.cfg.NodeInfo.NodeID.SQLInstanceID()
> 	const colsPerEvent = 5
> 	// Note: we insert the alue zero as targetID because sadly this
> ```

let's add a migration removing that constraint?
___
*[`pkg/sql/event_log.go` line 453 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ybpz5BVuEfllA5lKX:-N9Ybpz5BVuEfllA5lKY:b-d63shf) ([raw file](https://github.com/cockroachdb/cockroach/blob/383a2ab1ed12ca06132d02beb619f323460402aa/pkg/sql/event_log.go#L453)):*
> ```Go
> func insertEventRecords(
> 	ctx context.Context,
> 	execCfg *ExecutorConfig,
> ```

I'm not sure changing this param is an improvement. The main thing that's needed out of the `ExecutorConfig` is an `InternalExecutor`; for the rest of the needed things, the `ExecutorConfig` is not a very appropriate struct to pass. Between the two, I prefer it how it was.
___
*[`pkg/sql/event_log.go` line 406 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YcsXMBK11S__eVIr2:-N9YcsXMBK11S__eVIr3:b-hoihe) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L406)):*
> ```Go
> )
> 
> var eventLogAsyncWritesMaxAttempts = settings.RegisterIntSetting(
> ```

I'd vote we hardcode a number instead of this setting
___
*[`pkg/sql/event_log.go` line 414 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ycona3vBGo9FVXcJA:-N9Ycona3vBGo9FVXcJB:b6j7fzi) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L414)):*
> ```Go
> )
> 
> var eventLogAsyncWritesAttemptTimeout = settings.RegisterDurationSetting(
> ```

I'd vote we hardcode a timeout instead of this setting
___
*[`pkg/sql/event_log.go` line 461 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YcxQy94E0FewAVKCW:-N9YcxQy94E0FewAVKCX:bvf3kwo) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L461)):*
> ```Go
> 	// filtering. Easiest is to pretend the event is sourced here.
> 	//
> 	// NB: insertEventRecords never returns an error when provided a nil txn.
> ```

instead of this NB, assert that the error is nil and put the explanation in the panic
___
*[`pkg/sql/event_log.go` line 551 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YdMkK7xvp3BM58eGO:-N9YdMkK7xvp3BM58eGP:b-53m8so) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L551)):*
> ```Go
> 	}
> 	// No: do them async.
> 	txn.AddCommitTrigger(func(ctx context.Context) {
> ```

I'm confused; doesn't this crash if `txn` is nil?
___
*[`pkg/sql/event_log.go` line 564 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YdZpF-llLbPwILNGD:-N9YdZpF-llLbPwILNGE:b4h7b55) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L564)):*
> ```Go
> 	stopper := execCfg.RPCContext.Stopper
> 	return stopper.RunAsyncTask(
> 		context.Background(), "record-events", func(bgCtx context.Context) {
> ```

why not use `ctx`?
___
*[`pkg/sql/event_log.go` line 598 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YdoTL5yIN96Pz7vzs:-N9YdoTL5yIN96Pz7vzt:bd3j4cl) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L598)):*
> ```Go
> }
> 
> func synchronousWriteToSystemEventsTable(
> ```

nit: remove `synchronous` from the name

**Review by knz — COMMENTED** [review]
This is ready for another look.

NB: the very last commit is currently expected to fail CI due to #86132. I'll rebase when the fix in #86153 merges.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-:-N9_rtBu289JNlmBxTcX:b92w5s6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei)

___
*[`-- commits` line 4 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YZt-d9KBpNhP7-P8T):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

is it really?
</blockquote></details>

I've fixed the release justifications throughout.
___
*[`-- commits` line 19 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YZun98oQDNL6k6hv_):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

is it really?
</blockquote></details>

Done.
___
*[`-- commits` line 42 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YcOwt0PCzvXyXqwv5):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

"cluster setting"
</blockquote></details>

Done.
___
*[`-- commits` line 49 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ycdim1Sswr0g7deoK):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

do we really need this cluster setting? I'd vote against it.
</blockquote></details>

It's not public. We need it for certain logic tests (not all) which is why a testing knob won't do.
___
*[`pkg/sql/event_log.go` line 128 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Y_5NjBKmfvPstiE4v:-N9_hGUR0zxMzoeWovCk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6b7f0058325de8b8b321a08137d462c418c3047f/pkg/sql/event_log.go#L128)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

please put a comment somewhere around the schema definition for `system.eventlog` saying that the respective column is not populated any more (or is populated with 0)
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 286 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YalJi08MwqjCu2QHW:-N9_jze-E-kXTepAi7Q3:brt42tn) ([raw file](https://github.com/cockroachdb/cockroach/blob/f8a8de03713e02a3a0b7039908209054316ffc72/pkg/sql/event_log.go#L286)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I don't really understand what's going on here. Perhaps put a method comment explaining what should and should not be populated in `commonSQLEventDetails`.
</blockquote></details>

Added comments and extended the commit message to explain what's going on.
___
*[`pkg/sql/event_log.go` line 534 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Y__bDDNU8PYv-HFh5:-N9_memV2TVCPzfIKmqB:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/f8a8de03713e02a3a0b7039908209054316ffc72/pkg/sql/event_log.go#L534)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

alue
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 534 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Y_ne_2748fjgM7Van:-N9_mf4w5XjZ8tGhkN-U:b-7yd1q4) ([raw file](https://github.com/cockroachdb/cockroach/blob/f8a8de03713e02a3a0b7039908209054316ffc72/pkg/sql/event_log.go#L534)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

let's add a migration removing that constraint?
</blockquote></details>

Added a TODO.
___
*[`pkg/sql/event_log.go` line 453 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ybpz5BVuEfllA5lKX:-N9_lrEp-xc7hHWGpW86:b-t9ndrq) ([raw file](https://github.com/cockroachdb/cockroach/blob/383a2ab1ed12ca06132d02beb619f323460402aa/pkg/sql/event_log.go#L453)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I'm not sure changing this param is an improvement. The main thing that's needed out of the `ExecutorConfig` is an `InternalExecutor`; for the rest of the needed things, the `ExecutorConfig` is not a very appropriate struct to pass. Between the two, I prefer it how it was.
</blockquote></details>

It's mostly for the benefit of the next commit. I've explained in the commit message.
___
*[`pkg/sql/event_log.go` line 406 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YcsXMBK11S__eVIr2:-N9_k3U9DDJQM2901XDD:blmdbtj) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L406)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I'd vote we hardcode a number instead of this setting
</blockquote></details>

OK, Done.
___
*[`pkg/sql/event_log.go` line 414 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ycona3vBGo9FVXcJA:-N9_lJSk1OiAHnX-Vito:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L414)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I'd vote we hardcode a timeout instead of this setting
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 461 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YcxQy94E0FewAVKCW:-N9_lq2rBl3z--VXtu8L:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L461)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

instead of this NB, assert that the error is nil and put the explanation in the panic
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 551 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YdMkK7xvp3BM58eGO:-N9_lvE070IVZ-yI-uaU:b-i6gti3) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L551)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I'm confused; doesn't this crash if `txn` is nil?
</blockquote></details>

It did :sweat_smile: 
Ive fixed it since.
___
*[`pkg/sql/event_log.go` line 564 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YdZpF-llLbPwILNGD:-N9_m3lN1eSl-dipvqn2:b-i7kpb) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L564)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

why not use `ctx`?
</blockquote></details>

Because `ctx` is linked to the surrounding query's execution, and gets canceled when the execution stops. The async task writing the event to the log can run after the surrounding SQL statement has completed execution.
___
*[`pkg/sql/event_log.go` line 598 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9YdoTL5yIN96Pz7vzs:-N9_mEez4pJdh9gHqXeX:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a5884ee27ec8440e35a79f477678efb20d82f82c/pkg/sql/event_log.go#L598)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

nit: remove `synchronous` from the name
</blockquote></details>

Done.

**Review by andreimatei — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-:-N9bMpT77SXdPw5kJ4SH:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei and @knz)

___
*[`-- commits` line 49 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ycdim1Sswr0g7deoK):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

It's not public. We need it for certain logic tests (not all) which is why a testing knob won't do.
</blockquote></details>

you can use a `cluster-opt: ...` line in logic tests to set testing knobs (and add support to `logic.go` for a new knob). If the tests can get away with setting the knob for the whole duration of the test (and I think here they do). I fell like a knob here would be preferable (even to a non-public setting), but up to you.
___
*[`-- commits` line 69 at r13](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bMhZR9OywTQIP0ZXH):*
I don't think this should be a separate commit.
___
*[`pkg/sql/event_log.go` line 260 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bGTm3BVOS1Q8RRs5Q:-N9bGTm3BVOS1Q8RRs5R:b-8vw0h3) ([raw file](https://github.com/cockroachdb/cockroach/blob/72972c4a8cd924dca5e83796cf41af074ded4ec6/pkg/sql/event_log.go#L260)):*
> ```Go
> // have access to a (*planner) and the current statement metadata.
> //
> // It does so by synthetizing the (shared) CommonSQLEventDetails,
> ```

Now what's going on is more clear because of the code comment, but I don't understand what "synthesize" means here. This method takes in `commonSQLEventDetails`, it doesn't seem to synthesize anything. I'd remove this paragraph.
___
*[`pkg/sql/event_log.go` line 486 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bJryI4udtzIsLg84b:-N9bJryI4udtzIsLg84c:b-b46jxx) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L486)):*
> ```Go
> // the run-time type of the event payload.
> //
> // If the txn field is non-nil and the cluster setting
> ```

This comment is hard to parse, in my opinion. If `server.eventlog.sync_writes.enabled` didn't exist, there'd be no need for it :P
___
*[`pkg/sql/event_log.go` line 490 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bKM55375kptQ73XT7:-N9bK8I4A7QZb6k7oLyV:b-nb7k6y) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L490)):*
> ```Go
> // on the given txn object. Otherwise, an asynchronous task is spawned
> // to do the write.
> // If txn is nil, the event is written asynchronously.
> ```

even if `txn` is not nil, the write is still async (as a commit trigger), right? I think commit triggers should be mentioned in this comment.
___
*[`pkg/sql/event_log.go` line 585 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bKqgF1BZXrug8qQY7:-N9bKqgF1BZXrug8qQY8:bt9qju) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L585)):*
> ```Go
> 	stopper := execCfg.RPCContext.Stopper
> 	if err := stopper.RunAsyncTask(
> 		context.Background(), "record-events", func(bgCtx context.Context) {
> ```

If `origCtx` stays named `origCtx`, then you can rename `bgCtx` to `ctx`, which would be better in my opinion.
As a nit, I would also rename `origCtx` to `ctx` as the parameter name (from the perspective of the function's caller, `origCtx` doesn't make sense), and I'd do a `origCtx := ctx` before the `RunAsyncTask()` call.
___
*[`pkg/sql/event_log.go` line 585 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bMvWZ7O66B5IESum4:-N9bMvWZ7O66B5IESum5:b-x5e242) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L585)):*
> ```Go
> 	stopper := execCfg.RPCContext.Stopper
> 	if err := stopper.RunAsyncTask(
> 		context.Background(), "record-events", func(bgCtx context.Context) {
> ```

please put a comment around here saying that we don't want to inherit the caller's cancelation
___
*[`pkg/sql/event_log.go` line 620 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bLq2RE7i_rT1r8sXu:-N9bLq2RE7i_rT1r8sXv:b-343yrm) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L620)):*
> ```Go
> 			}
> 		}); err != nil {
> 		log.Ops.Warningf(origCtx, "failed to start task to log %d events: %v", len(entries), err)
> ```

I don't think using the `Ops` channel here (as opposed to simply `log.Infof` makes sense). In fact, I wouldn't log this error at all, and definitely not at the warning level. If the server is shutting down and we can't write to the system table any more, I think it's just bad luck; I don't think anyone will benefit from these log message.
Same for the Warning above too.

**Review by knz — COMMENTED** [review]
I've merged the last two commits as recommended.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-:-N9m4mCU-KPwy1RojB4c:b-yfiujw)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andreimatei)

___
*[`-- commits` line 49 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9Ycdim1Sswr0g7deoK):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

you can use a `cluster-opt: ...` line in logic tests to set testing knobs (and add support to `logic.go` for a new knob). If the tests can get away with setting the knob for the whole duration of the test (and I think here they do). I fell like a knob here would be preferable (even to a non-public setting), but up to you.
</blockquote></details>

Done.
___
*[`-- commits` line 69 at r13](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bMhZR9OywTQIP0ZXH):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I don't think this should be a separate commit.
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 260 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bGTm3BVOS1Q8RRs5Q:-N9lzLkI7A3zRvqN2Wb1:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/72972c4a8cd924dca5e83796cf41af074ded4ec6/pkg/sql/event_log.go#L260)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

Now what's going on is more clear because of the code comment, but I don't understand what "synthesize" means here. This method takes in `commonSQLEventDetails`, it doesn't seem to synthesize anything. I'd remove this paragraph.
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 486 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bJryI4udtzIsLg84b:-N9m47C1Cz9Ei02OXRpg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L486)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

This comment is hard to parse, in my opinion. If `server.eventlog.sync_writes.enabled` didn't exist, there'd be no need for it :P
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 490 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bKM55375kptQ73XT7:-N9m47mh1glxn3_bQWzD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L490)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

even if `txn` is not nil, the write is still async (as a commit trigger), right? I think commit triggers should be mentioned in this comment.
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 585 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bKqgF1BZXrug8qQY7:-N9lvj9r7-oGamO3Oa3f:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L585)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

If `origCtx` stays named `origCtx`, then you can rename `bgCtx` to `ctx`, which would be better in my opinion.
As a nit, I would also rename `origCtx` to `ctx` as the parameter name (from the perspective of the function's caller, `origCtx` doesn't make sense), and I'd do a `origCtx := ctx` before the `RunAsyncTask()` call.
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 585 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bMvWZ7O66B5IESum4:-N9lvjuP6gm1bCo1fUxH:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L585)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

please put a comment around here saying that we don't want to inherit the caller's cancelation
</blockquote></details>

Done.
___
*[`pkg/sql/event_log.go` line 620 at r12](https://reviewable.io/reviews/cockroachdb/cockroach/86174#-N9bLq2RE7i_rT1r8sXu:-N9lvlYw7M05YXiC8BxC:b-q1qi04) ([raw file](https://github.com/cockroachdb/cockroach/blob/b32f831298a44db0ecb6a9b7dbac6b4d0bc00a59/pkg/sql/event_log.go#L620)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

I don't think using the `Ops` channel here (as opposed to simply `log.Infof` makes sense). In fact, I wouldn't log this error at all, and definitely not at the warning level. If the server is shutting down and we can't write to the system table any more, I think it's just bad luck; I don't think anyone will benefit from these log message.
Same for the Warning above too.
</blockquote></details>

The one above is definitely useful because perhaps the eventlog table is unavailable, and we want to see symptoms of that.

For the one here, rephrased as an assertion failure. This ensures we get a sentry report for it if it's ever hit.

**Comment by knz** [comment]
TFYR!

bors r=andreimatei

**Comment by knz** [comment]
bors r=andreimatei

**Comment by knz** [comment]
bors r=andreimatei

**Comment by ajwerner** [comment]
>pkg/sql/schemachanger/scdeps/sctestdeps/test_deps.go:1024:9: cannot use s (variable of type *TestState) as type scexec.EventLogger in return statement:
[22:42:27 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_UnusedLint/6173817?buildTab=log&focusLine=5074&linesState=5074)        *TestState does not implement scexec.EventLogger (wrong type for LogEvent method)
[22:42:27 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_UnusedLint/6173817?buildTab=log&focusLine=5075&linesState=5075)          have LogEvent(_ context.Context, descID catid.DescID, details eventpb.CommonSQLEventDetails, event logpb.EventPayload) error
[22:42:27 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_UnusedLint/6173817?buildTab=log&focusLine=5076&linesState=5076)          want LogEvent(ctx context.Context, details eventpb.CommonSQLEventDetails, event logpb.EventPayload) error

bors r-

**Comment by ajwerner** [comment]
Race condition. You got. Sorry :)

**Comment by knz** [comment]
@ajwerner what is the PR/branch?

**Comment by knz** [comment]
nvm found it

**Comment by knz** [comment]
bors r=andreimatei
